src/core/FileManagerServices.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * FileManagerServices.cpp - Helper file for cross platform file management
+ *
+ * Copyright (c) 2025 Andrew Wiltshire <aw1lt / at/ proton/ dot/me >
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "FileManagerServices.h"
+
+#include <QDesktopServices>
+#include <QDir>
+#include <QProcess>
+#include <QUrl>
+
+namespace lmms {
+void FileManagerServices::openDir(QString& path)
+{
+	QString nativePath = QDir::toNativeSeparators(path);
+
+	QProcess::startDetached(getDefaultFileManager(), {nativePath});
+}
+void FileManagerServices::reveal(const QFileInfo item)
+{
+	if (!canSelect()) { QDesktopServices::openUrl(QUrl::fromLocalFile(item.absolutePath())); }
+	QString path = QDir::toNativeSeparators(item.canonicalFilePath());
+	QStringList params;
+
+#ifdef _WIN32
+	// explorer /select,[object]: Selects the file in the new explorer window
+	params += QLatin1String("/select,");
+#elif __APPLE__
+	// Finder -R, --reveal: Selects in finder
+	params += "-R";
+#else
+	// --select: Linux, BSD and other *nix systems
+	params += "--select";
+#endif
+	params += path;
+	QProcess::startDetached(getDefaultFileManager(), params);
+}
+
+bool FileManagerServices::supportsSelectOption(const QString& fileManager)
+{
```

Comment:
I believe this function should use the same cache-checking technique as `getDefaultFileManager` so that reentry only causes it to run once.